kuma/users/views.py

```diff
@@ -584,12 +595,14 @@ def form_valid(self, form):
     def get_context_data(self, **kwargs):
         context = super(SignupView, self).get_context_data(**kwargs)
 
-        # For GitHub users, find matching legacy Persona social accounts
-        assert self.sociallogin.account.provider == 'github'
+        # For GitHub/Google users, find matching legacy Persona social accounts
+        assert self.sociallogin.account.provider in ('github', 'google')
         or_query = []
         for email_address in self.email_addresses.values():
             if email_address['verified']:
-                or_query.append(Q(uid=email_address['email']))
+                # only persona accounts have emails as UIDs
+                # but adding the provider criteria makes this explicit and future-proof
+                or_query.append(Q(uid=email_address['email']) & Q(provider='persona'))
```

Comment:
Nit: This multi-line chunk could be simplified to:
```python
uids = Q()
for email_address in self.email_addresses.values():
    if email_address['verified']:
        uids |= Q(uid=email_address['email'])
if uids:
    matching_accounts = SocialAccount.objects.filter(uids, provider='persona')
else:
    ...
```

Reply:
Thanks, this is much better indeed!